src/probnum/problems/_problems.py

```diff
@@ -1 +1,157 @@
+    >>> rp.observations
+    [11.4123, -15.5123]
+    """
+
+    observations: np.ndarray
+    locations: np.ndarray
+    likelihood: "probnum.filtsmooth.statespace.DiscreteGaussian" = None
+
+
+@dataclasses.dataclass
+class IVProblem:
+    r"""First order ODE initial value problem.
+
+    Compute a function :math:`y=y(t)` that solves
+
+    .. math::
+        \dot y(t) = f(t, y(t)), \quad y(t_0) = y_0
+
+    on time-interval :math:`[t_0, t_\text{max}]`.
+    Solved by the ODE solvers in :mod:`probnum.diffeq`.
+
+
+    Parameters
+    ----------
+    f
+        ODE vector-field.
+    t0
+        Initial point in time.
+    tmax
+        Final point in time.
+    y0
+        Initial value of the solution.
+
+    Examples
+    --------
+    >>> def f(t, x):
+    ...     return x*(1-x)
+    >>> ivp = IVProblem(f, t0=0., tmax=3., y0=0.1)
+    >>> ivp.t0, ivp.tmax, ivp.y0
+    (0.0, 3.0, 0.1)
+    >>> np.round(ivp.f(ivp.t0, ivp.y0), 2)
+    0.09
+    """
+
+    f: typing.Callable[[float, np.ndarray], np.ndarray]
+    t0: float
+    tmax: float  # Bold move: remove this? This is not really part of an IVP.
+    y0: typing.Union[probnum.type.FloatArgType, np.ndarray]
+
+
+@dataclasses.dataclass
+class LinearSystemProblem:
+    r"""Linear system of equations.
+
+    Compute :math:`x` from :math:`Ax=b`.
+    Solved by the probabilistic linear solver in mod:`probnum.linalg`
+
+    Parameters
+    ----------
+    A
+        Square matrix or linear operator.
+    b
+        Right-hand side vector or matrix.
+
+    Examples
+    --------
+    >>> A = np.eye(3)
+    >>> b = np.arange(3)
+    >>> lsp = LinearSystemProblem(A, b)
+    >>> lsp
+    LinearSystemProblem(A=array([[1., 0., 0.],
+           [0., 1., 0.],
+           [0., 0., 1.]]), b=array([0, 1, 2]))
+    """
+
+    A: typing.Union[
+        np.ndarray,
+        "scipy.sparse.spmatrix",
+        "probnum.linops.LinearOperator",
+        "probnum.RandomVariable",
+    ]
+    b: typing.Union[np.ndarray, "probnum.RandomVariable"]
+
+
+@dataclasses.dataclass
+class QuadProblem:
+    r"""Numerical computation of an integral.
+
+    Compute the integral
+
+        .. math::
+            \int_\Omega f(x) \, \text{d} \mu(x)
+
+    for a function :math:`f: \Omega \rightarrow \mathbb{R}`.
+    For the time being, :math:`\mu` is the Lebesgue measure.
+    Solved by the quadrature rules in :mod:`probnum.quad`.
+
+    Parameters
+    ----------
+    integrand
+        Function to be integrated.
+    domain
+        Domain of the integral.
+
+    Examples
+    --------
+    >>> def integrand(x):
+    ...     return x**2
+    >>> def domain(x):
+    ...     return 0 < x < 1
+    >>> qp = QuadProblem(integrand, domain)
+    >>> np.round(qp.integrand(0.2), 2)
+    0.04
+    >>> qp.domain(0.2)
+    True
+    """
+
+    integrand: typing.Callable[[np.ndarray], np.ndarray]
+    domain: typing.Callable[[np.ndarray], bool]  # Up for discussion...
```

Comment:
@alpiges this is only a draft -- I don't know how you usually pick domains in code.

Reply:
An alternative would be something similar to (https://github.com/SciML/Quadrature.jl  --  see bottom of the readme). Let me know if this is more in line with your problem types [or change it yourself in the future :)]

Reply:
I think the domain should be defined as box constraints (lower and upper bound for each dimension of x). Although there could be a domain method which checks whether x is in the domain. Just -- why?
Also the example domain will not work in multiple dimensions.

Reply:
> I think the domain should be defined as box constraints

Thanks a lot for the feedback, I will change it! :) 

> Just -- why?

I thought box constraints only work for rectangular (do I mean "hypercubic" or something like this?) domains. What about triangles, circles, L-shaped, domains, etc.? (see for instance https://github.com/nschloe/quadpy#line-segment-c1) 

>  Also the example domain will not work in multiple dimensions.

I agree. It is just a simple example... :)

Reply:
> I thought box constraints only work for rectangular (do I mean "hypercubic" or something like this?) domains. What about triangles, circles, L-shaped, domains, etc.? (see for instance https://github.com/nschloe/quadpy#line-segment-c1)
Yes, I agree, just that's pretty far on the horizon. As long as we can to kernel integrals in closed form, we need the explicit bounds of the domain.